tests/kw_db_test.sh

```diff
@@ -70,11 +70,86 @@ function test_execute_sql_script()
   }
 }
 
+function test_format_entries_db()
+{
+  local output
+  local expected
+  local ret
+
+  cd "$FAKE_DATA" || {
+    ret="$?"
+    fail "($LINENO): Failed to move to fake data dir"
+    exit "$ret"
+  }
+
+  output=$(format_entries_db)
+  ret="$?"
+  expected='No arguments given'
+  assert_equals_helper 'No arguments, should fail' "$LINENO" 22 "$ret"
+  assert_equals_helper 'Expected error msg' "$LINENO" "$output" "$expected"
+
+  output=$(format_entries_db 'first' 'second' 'third')
+  ret="$?"
+  expected='first,second,third'
+  assert_equals_helper 'No error expected' "$LINENO" 0 "$ret"
+  assert_equals_helper 'Wrong output' "$LINENO" "$output" "$expected"
+
+  cd "$ORIGINAL_DIR" || {
+    ret="$?"
+    fail "($LINENO): Failed to move back from fake data dir"
+    exit "$ret"
+  }
+}
+
+function test_format_values_db()
+{
+  local output
+  local expected
+  local ret
+
+  cd "$FAKE_DATA" || {
+    ret="$?"
+    fail "($LINENO): Failed to move to fake data dir"
+    exit "$ret"
+  }
+
+  output=$(format_values_db 0)
+  ret="$?"
+  expected='No arguments given'
+  assert_equals_helper 'Invalid db, error expected' "$LINENO" 22 "$ret"
+  assert_equals_helper 'Expected error msg' "$LINENO" "$output" "$expected"
+
+  output=$(format_values_db 3 'first' 'second' 'third')
+  ret="$?"
+  expected="('first','second','third')"
+  assert_equals_helper 'No error expected' "$LINENO" 0 "$ret"
+  assert_equals_helper 'Wrong output' "$LINENO" "$output" "$expected"
+
+  output=$(format_values_db 2 'first1' 'second1' 'first2' 'second2')
+  ret="$?"
+  expected="('first1','second1'),('first2','second2')"
+  assert_equals_helper 'No error expected' "$LINENO" 0 "$ret"
+  assert_equals_helper 'Wrong output' "$LINENO" "$output" "$expected"
+
+  output=$(format_values_db 1 "some 'quotes'")
+  ret="$?"
+  expected="('some ''quotes''')"
```

Comment:
Don't we have one extra quote here?

Reply:
Nope. There should be quotes around the full string, and then 1 extra quote for each quote inside the original string.